app/models/reservation.rb

```diff
@@ -17,4 +17,17 @@ class Reservation < ApplicationRecord # :nodoc:
   validates :checkin_at, presence: true
   validates :checkout_at, presence: true
   validates :book_code, presence: true, uniqueness: true
+
+  # Callbacks
+  before_create :assign_available_room, if: -> { room.status == 'unavailable' }
```

Comment:
`enum` comes with a bunch of predicate method so you don't need to check against raw string e.g.
```suggestion
  before_create :assign_available_room, if: -> { room.status.unavailable? }
```

Reply:
Actually, I think you can even do:

`before_create :assign_available_room, if: -> { room.unavailable? }`